src/main.cpp

```diff
@@ -783,6 +793,31 @@ int main ( int argc, char** argv )
                 strServerBindIP = "";
             }
         }
+
+#ifndef NO_JSON_RPC
+        //
+        // strJsonRpcBind address defaults to loopback and should not be empty, but
+        // in the odd chance that an empty IP is passed, we'll check for it here.
+
+        if ( strJsonRpcBindIP.trimmed().isEmpty() )
+        {
+            qCritical() << qUtf8Printable ( QString ( "JSON-RPC is enabled but no bind address was provided, exiting." ) );
+            exit ( 1 );
+        }
+        else
+        {
+            // This means of testing the validity of IP addresses is far from perfect but
+            // we do it here as an upfront check.  The downstream network calls will error
+            // out on malformed addresses not caught here.
+
+            QHostAddress InetAddr;
+            if ( !InetAddr.setAddress ( strJsonRpcBindIP ) )
+            {
+                qCritical() << qUtf8Printable ( QString ( "The JSON-RPC address specified is not valid, exiting. " ) );
+                exit ( 1 );
+            }
+        }
```

Comment:
We can avoid deep nesting here by dropping `else` (we'll never reach that code due to the `exit` before.

Reply:
I'll refactor taking into account @pljones comment about the port being specified as well.

Reply:
FWIW, the reason this was in an if..else block was to manage the variable scope for InetAddr.  Even if I pull it out and make the "else" its own block, I need to (or should) bracket it to protect that scope.